app/tests/reader_studies_tests/test_views.py

```diff
@@ -384,6 +384,39 @@ def test_display_set_update_permissions(client):
     assert response.status_code == 200
 
 
+@pytest.mark.django_db
+def test_display_set_detail_permissions(client):
+    rs = ReaderStudyFactory()
+    rs.add_editor(editor := UserFactory())
+    rs.add_reader(reader := UserFactory())
```

Comment:
A line costs nothing, and it is incredibly weird to have what looks like a function call have side effects in the parent frame. This is also subtly not using kwargs. Please read https://peps.python.org/pep-0572/#the-importance-of-real-code, and especially "Exceptional cases", for appropriate uses of this operator.

Reply:
Noted. Looking back at this introduction I think I know why I found the single line attractive, despite now agreeing it is too confusing to read:

```Python
editor = UserFactory()
rs.add_editor(editor)
```

On the first line, it isn't an editor yet...